exchange-v2/contracts/ExchangeV2Core.sol

```diff
@@ -243,12 +243,19 @@ abstract contract ExchangeV2Core is Initializable, OwnableUpgradeable, AssetMatc
     }
 
     /// @dev Checks if order is fully filled, if true then deletes it
-    function deleteFilledOrder(LibOrder.Order memory order, bytes32 hash) internal {
+    function deleteFilledOrder(LibOrder.Order memory order, bytes32 hash, LibOrderDataV2.DataV2 memory dataOrder) internal {
         if (!isTheSameAsOnChain(order, hash)) { 
             return;
         }
 
-        uint takeValueLeft = order.takeAsset.value.sub(getOrderFill(order, hash));
+        uint value;
+        if (dataOrder.isMakeFill) {
+            value = order.makeAsset.value;
+        } else {
+            value = order.takeAsset.value;
+        }
+
+        uint takeValueLeft = value.sub(getOrderFill(order, hash));
```

Comment:
deleteFilledOrder() can take fills of orders as argument (LibFill.FillResult memory newFill from matchAndTransfer() ) so we don't read storage 2 times

Reply:
method getFillSetNew() read **fills** and remember it in return value, after change **fills**
getOrderFill() don`t need old **fills**, it use new one
so there are a lot of broken tests if use LibFill.FillResult memory newFill from matchAndTransfer()